Source/WebCore/rendering/RenderListItem.cpp

```diff
@@ -231,25 +231,47 @@ void RenderListItem::updateValueNow() const
         }
     }
 
+    int defaultIncrement = (orderedList && orderedList->isReversed()) ? -1 : 1;
+    // FIXME: This should account for counter-reset as well.
+    auto valueForItem = [&](int previousValue, CounterDirectives& directives) {
+        if (directives.setValue)
+            return *directives.setValue;
+        return previousValue + directives.incrementValue.value_or(defaultIncrement);
+    };
+
     auto& startValue = startItem->m_value;
-    if (!startValue)
-        startValue = orderedList ? orderedList->start() : 1;
+    if (!startValue) {
+        auto directives = startItem->style().counterDirectives().map.get("list-item"_s);
```

Comment:
This copies the `CounterDirectives` object. Though at 16 bytes it's not a big deal.

Reply:
will use `auto&` if possible.

Reply:
I was trying to read the `HashMap` and `HashTraits` code to work out if `get()` returned a referenced or a copy of the data, but wasn't sure without building.

Reply:
The `get` function returns a copy, not a reference to the value in the map, so it can handle values not in the map. If you want to avoid a copy you have to use the wordier `find`/`end` combination.

Reply:
I think that in this case a copy may be a good thing. A small scalar that likely ends up in registers.